### DISCUSSION TITLE
Welcome to go-peer Discussions!

### DISCUSSION THREAD

**Discussion by number571** [number571/go-peer#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by garrytourist** [comment]
How can I run this on my device and test it? . I downloaded and unzipped it myself
The command from the instructions doesn't work ( go installed) ) .

**Reply by number571** [reply]
Try this way:
```bash
$ go mod init gotest
$ go get github.com/number571/go-peer
$ go mod tidy
```

as example go file:
```go
package main

import (
	"fmt"

	"github.com/number571/go-peer/pkg/encoding"
)

func main() {
	fmt.Println(encoding.HexEncode([]byte("hello")))
}
```

**Reply by garrytourist** [reply]
thank you for your answer. I finally managed to launch your network application. I am slowly studying and reading your articles. I am very grateful to you for your contribution.

I tried to write to you using the key in your network but I must have misunderstood something and I am not getting a response.
In order for me to receive a response, you must have my public key. Will it be sent to you automatically when I try to write to you or do I need to send it to you separately?

maybe I could chat with you in some application for this?

**Reply by number571** [reply]
Yes, I also need to get your public key to establish a connection. You can send your public key via any convenient communication channel that you trust. As an example, you can register your public key here: https://github.com/number571/hidden-public-keys.

**Reply by garrytourist** [reply]
Unfortunately I don't have access to change the repository(

**Reply by number571** [reply]
You can fork the repository, insert your public key, and send a pull request. I will already be able to add hashes of this public key in the README file.

**Reply by garrytourist** [reply]
I'll leave the key and checksums here. If you have a time, add them to the repository and if possible, write me something on your network.

```
sha256 5c3d59a77161893122788eddcc073dd0e5717b1c22c43b9274c15d9d49ab2305
sha384 66b2b002c6de2e350fe6f106b012286015f3e222dc00c8df63ba8605d2e452da6f122eeefd917d51105675772a5e8df4
sha512 23f8d4a3a1e3c2209d4f26ac4d223fe645d0064e4ce5c62f60207508b1a41ae78e5d484b83c8798badc5d76d352c9cb4e4fd02f46eb2ddec8ddf40e04d298570
```

archive sum 
[firstCon.zip](https://github.com/user-attachments/files/18708821/firstCon.zip)
sha256 852e33a869674895d489cc5b74f02f0db69bd7e3bb6e93b0b73d52f9dd9427bb  firstCon.zip

```
[firstCon.zip](https://github.com/user-attachments/files/18708829/firstCon.zip)

**Reply by number571** [reply]
Done. I also added a public key to my friends list.

**Reply by garrytourist** [reply]
I am waiting for your message.
